operator/v1/types_console.go

```diff
@@ -106,6 +106,36 @@ type ConsoleCustomization struct {
 	// SVG format preferred
 	// +optional
 	CustomLogoFile configv1.ConfigMapFileReference `json:"customLogoFile,omitempty"`
+	// console allow the cluster admin to configure the shown categories.
+	DeveloperCatalog DeveloperConsoleCatalogCustomization `json:"developerCatalog"`
+}
+
+// ConsoleCatalog allow the cluster admin to configure developer catalog.
+type DeveloperConsoleCatalogCustomization struct {
+	// categories which are shown the in the devconsole.
+	Categories []DeveloperConsoleCatalogCategory `json:"categories"`
+}
+
+type DeveloperConsoleCatalogCategory struct {
```

Comment:
I think we can simplify this by:
```go
// proper comment here
type DeveloperConsoleCatalogCategory struct {
        // proper comment here
	DeveloperConsoleCatalogMeta `json:",inline"`
	// a list of child categories
	// +optional
	Subcategories []DeveloperConsoleCatalogMeta `json:"subcategories,omitempty"`
}

// proper comment here
type DeveloperConsoleCatalogMeta struct {
	// identifier used in the URL to enable deeplinking in console
	ID string `json:"id"`
	// category display label
	Label string `json:"label"`
	// the list of item tags this category will match
	// +optional
	Tags []string `json:"tags"`
}
```

Please add proper comments as well

Reply:
Ah thanks, didn't know that. I got some issues with loops but this looks goot, will test this.